File: 7kyu/doubleEveryOther.js
/*
Write a function that doubles every second integer in a list, starting from the left.
*/

function doubleEveryOther(a) {
    return a.map((n, i) => i % 2 === 0 ? n : n * 2);
}

console.log(doubleEveryOther([1, 2, 3, 4])) // [1,4,3,8]

File: 7kyu/double_every_other.py
# Write a function that doubles every second integer in a list, starting from the left.


def double_every_other(a):
    result = []

    for idx, n in enumerate(a):
        if idx % 2 != 0:
            result.append(n * 2)
        else:
            result.append(n)

    return result


print(double_every_other([1, 2, 3, 4]))  # [1,4,3,8]
